docs/cow-protocol/tutorials/arbitrate/driver.md

```diff
@@ -3,3 +3,90 @@ sidebar_position: 1
+You can even merge the responsibilities of the driver and matching engine into one if you want to build the most optimal solver possible.  
+The only hard requirement is that the component the autopilot interfaces with implements this [interface](api/driver).
+
+### Responsibilities
+
+#### Preprocessing Auctions
+
+The auctions sent to the driver by the autopilot only contain the bare minimum of information needed.
+But usually a matching engine requires more information than that so the driver pre-processes the auction before forwarding it to the matching engine.
+That process includes:
+* fetching additional meta data (e.g. token decimals)
+* discarding orders that can definitely not be settled (e.g. user is missing balances)
+* very basic prioritization of remaining orders (e.g. orders below or close to the market price are most likely to settle)
+
+#### Fetching Liquidity
+
+Unless you find a perfect CoW you'll need some sort of liquidity to settle an order with.
+To get your solver started with a good set of base liquidity the driver is able to index and encode a broad range of fundamental AMMs.
+These include `UniswapV2` and it's derivatives, `UniswapV3` as well as several liquidity sources from the `BalancerV2` family.
+All of these can be individually configured or completely disabled if your matching engine manages liquidity on its own.
+
+#### Postprocessing Solutions
+
+The driver expects the matching engine to return a recipe on how to solve a set of orders but the recipe itself could not be submitted on-chain.
+In the post-processing step the driver applies multiple sanity checks to the solution, encodes it into a transaction that could be executed on-chain and verifies that it actually simulates successfully before it considers the solution valid.
+All this is done because solvers can get slashed for misbehaving so the reference driver checks all it can to reduce the risk of running a solver as much as possible.
+If your matching engine returns multiple valid solutions the driver will pick the best one and only report that one to the autopilot to maximize the solver's chance of winning the auction.
+
+#### Submitting Settlements
+
+If the solver provided the best solution the autopilot will tell the driver to actually submit it on-chain.
+This is not as straight forward as it sounds when the blockchain is congested and liquidity sources are volatile.
+To protect the solver from losing ETH by submitting solutions that revert the driver continuously verifies that the solution still works while it submits it on-chain.
+As soon as it would revert the driver cancels the transaction to cut the losses to a minimum.
+
+## Methodology
+
+As you can see the driver has many responsibilities and discussing all of them in detail would be beyond the scope of this documentation but it's worth mentioning one guiding principle that applies to most of them:  
```

Comment:
Reading it, it felt like I was left hanging because no clear guiding principle followed. This is my recap of the next paragraph in "guiding principle" format. 

```suggestion
As you can see the driver has many responsibilities and discussing all of them in detail would be beyond the scope of this documentation but it's worth mentioning one guiding principle that applies to most of them: make the driver do as less work as possible in the hot path when processing an auction.
```